lib/rails_failover/active_record/handler.rb

```diff
@@ -64,7 +64,8 @@ def initiate_fallback_to_primary
                 spec_name,
                 role: handler_key,
               )
-            connection_active = connection.active?
+
+            connection_active = connection.verify!
```

Comment:
Actual fix is here.

Reply:
We should probably keep the call to `connection.active?` and use `.tap(&:verify!)` on line 66 because of the difference in implementations of `#verify!`. Indeed, with Rails 7.1, this will return `true` but the code from 7.0 and 6.1 is the following:
```ruby
reconnect! unless active?
```
So here the result from `reconnect!` could be something else than a truthy value and if, for whatever reason, the connection is already active, it will return `nil`.

Reply:
Hmm I'm thinking if we should just rely on a more reliable API where the implementation does not differ between versions. Executing a query seems to be the most stable IMO.

Reply:
`#verify!` is a reliable API. It will establish the connection if it’s necessary, and does nothing if it’s already there. Then you just call `#active?` and it will work as expected (under the hood it’s executing a query).
We should not rely on `#verify!` to give us a meaningful return value, that’s all :slightly_smiling_face:

Reply:
Sounds good 👍 I've updated the PR.